flake.nix

```diff
@@ -8,183 +8,152 @@
-          passthru =
-            helix-wrapped.passthru
-            // {
-              wrapper = old: makeOverridableHelix old config;
-            };
-        };
-      stdenv =
-        if pkgs.stdenv.isLinux
-        then pkgs.stdenv
-        else pkgs.clangStdenv;
-      rustFlagsEnv = pkgs.lib.optionalString stdenv.isLinux "-C link-arg=-fuse-ld=lld -C target-cpu=native -Clink-arg=-Wl,--no-rosegment --cfg tokio_unstable";
-      rustToolchain = pkgs.pkgsBuildHost.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
-      craneLibMSRV = (crane.mkLib pkgs).overrideToolchain rustToolchain;
-      craneLibStable = (crane.mkLib pkgs).overrideToolchain pkgs.pkgsBuildHost.rust-bin.stable.latest.default;
-      commonArgs = {
-        inherit stdenv;
-        inherit (craneLibMSRV.crateNameFromCargoToml {cargoToml = ./helix-term/Cargo.toml;}) pname;
-        inherit (craneLibMSRV.crateNameFromCargoToml {cargoToml = ./Cargo.toml;}) version;
-        src = filteredSource;
-        # disable fetching and building of tree-sitter grammars in the helix-term build.rs
-        HELIX_DISABLE_AUTO_GRAMMAR_BUILD = "1";
-        buildInputs = [stdenv.cc.cc.lib];
-        nativeBuildInputs = [pkgs.installShellFiles];
-        # disable tests
-        doCheck = false;
-        meta.mainProgram = "hx";
+
+      # Get Helix's MSRV toolchain to build with by default.
+      msrvToolchain = pkgs.pkgsBuildHost.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+      msrvPlatform = pkgs.makeRustPlatform {
+        cargo = msrvToolchain;
+        rustc = msrvToolchain;
       };
-      cargoArtifacts = craneLibMSRV.buildDepsOnly commonArgs;
     in {
       packages = {
-        helix-unwrapped = craneLibStable.buildPackage (commonArgs
-          // {
-            cargoArtifacts = craneLibStable.buildDepsOnly commonArgs;
-            postInstall = ''
-              mkdir -p $out/share/applications $out/share/icons/hicolor/scalable/apps $out/share/icons/hicolor/256x256/apps
-              cp contrib/Helix.desktop $out/share/applications
-              cp logo.svg $out/share/icons/hicolor/scalable/apps/helix.svg
-              cp contrib/helix.png $out/share/icons/hicolor/256x256/apps
-              installShellCompletion contrib/completion/hx.{bash,fish,zsh}
-            '';
-            # set git revision for nix flake builds, see 'git_hash' in helix-loader/build.rs
-            HELIX_NIX_BUILD_REV = self.rev or self.dirtyRev or null;
-          });
-        helix = makeOverridableHelix self.packages.${system}.helix-unwrapped {};
+        # Make MSRV Helix
+        helix = pkgs.callPackage mkHelix {rustPlatform = msrvPlatform;};
+
+        # The default Helix build. Uses the default MSRV Rust toolchain, and the
+        # default nixpkgs, which is the one in the Flake.lock of Helix.
+        #
+        # This can be overridden though to add Cargo Features, flags, and different toolchains.
         default = self.packages.${system}.helix;
```

Comment:
add a `rec` on the `packages` attribute set and we can avoid the `self` ref by simply doing:

```nix
packages = rec {
    default = helix;
    helix = ...;
}
```

Reply:
I generally don't like doing that on my projects but this is simple enough that it should be fine.